base_name_search_improved/models/ir_model.py

```diff
@@ -29,25 +63,82 @@ def _extend_name_results(self, domain, results, limit):
         results.extend(recs.name_get())
     return results
 
+# TODO move all this to register_hook
+
+
+_add_magic_fields_original = models.BaseModel._add_magic_fields
+
+
+@api.model
+def _add_magic_fields(self):
+    res = _add_magic_fields_original(self)
+
+    if 'base_name_search_improved' in self.env.registry._init_modules:
+        if 'smart_search' not in self._fields:
```

Comment:
What about use `and` instead of two nested `if` in order to reduce cyclomatic complexity